dace/transformation/dataflow/redundant_array.py

```diff
@@ -200,11 +250,36 @@ def apply(self, sdfg):
         e1 = graph.edges_between(in_array, out_array)[0]
         a1_subset, b_subset = _validate_subsets(e1, sdfg.arrays)
 
+        # Find extraneous A or B subset dimensions
+        a_dims_to_pop = []
+        b_dims_to_pop = []
+        bset = b_subset
+        popped = []
+        if a1_subset and b_subset and a1_subset.dims() != b_subset.dims():
+            a_size = a1_subset.size_exact()
+            b_size = b_subset.size_exact()
+            if a1_subset.dims() > b_subset.dims():
+                a_dims_to_pop = find_dims_to_pop(a_size, b_size)
+            else:
+                b_dims_to_pop = find_dims_to_pop(b_size, a_size)
+                bset, popped = pop_dims(b_subset, b_dims_to_pop)
+
+        reduction = False
+        for e in graph.in_edges(in_array):
+            if isinstance(e.src, Reduce):
+                reduction = True
+
         # If the memlet does not cover the removed array, create a view.
-        if any(m != a for m, a in zip(a1_subset.size(), in_desc.shape)):
+        if reduction or any(m != a
```

Comment:
@tbennun Due to the improvements in RedundantArray, we can now automatically remove the transient between a Reduce node and its output when `keepdims=True`. However, the Reduce node does not handle this case well (the produced code is wrong). I found how to fix it in the pure expansion, but I guess that the same problem exists in all other implementations. Therefore, I opted to put a special case here; just change the transient to a view.